path_config.go

```diff
@@ -122,34 +121,31 @@ func (b *kubeAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Requ
 	}
 
 	disableLocalJWT := data.Get("disable_local_ca_jwt").(bool)
-	localCACert := []byte{}
-	localTokenReviewer := []byte{}
-	if !disableLocalJWT {
-		localCACert, _ = ioutil.ReadFile(localCACertPath)
-		localTokenReviewer, _ = ioutil.ReadFile(localJWTPath)
-	}
 	pemList := data.Get("pem_keys").([]string)
 	caCert := data.Get("kubernetes_ca_cert").(string)
 	issuer := data.Get("issuer").(string)
 	disableIssValidation := data.Get("disable_iss_validation").(bool)
-	if len(pemList) == 0 && len(caCert) == 0 {
```

Comment:
One benefit is that restoring Vault backup on another cluster would be possible.  Lets say, user ends up restoring Vault in a freshly deployed cluster as part of disaster recovery.  It wouldn't work since we stored wrong CA cert persistently in config.  

I think both `token` and `ca.crt` were never meant to be copied off the pod and persisted as part application config.  Instead, both are intended to be treated as injected data that Kubernetes API clients should read directly from disk when needed. This is how official Kubernetes API clients also behave when running in "in-cluster" mode.  It would be great if Vault would have the same behavior. Currently the docs give an impression that this is already the case https://www.vaultproject.io/api-docs/auth/kubernetes#caveats
..

Reply:
Hi @tomhjp! Have you had an opportunity to look at this yet, would it be acceptable to you to keep the `ca.crt` reloading as well with this reasoning?